src/audio/kpb.c

```diff
@@ -1130,9 +1135,13 @@ static void kpb_init_draining(struct comp_dev *dev, struct kpb_client *cli)
 		kpb->draining_task_data.dev = dev;
 		kpb->draining_task_data.sync_mode_on = kpb->sync_draining_mode;
 
-		/* Set host-sink copy mode to blocking */
-		comp_set_attribute(kpb->host_sink->sink, COMP_ATTR_COPY_TYPE,
-				   &copy_type);
+		/* save current sink copy type */
+		comp_get_attribute(kpb->host_sink->sink, COMP_ATTR_COPY_TYPE,
+				   &kpb->draininig_task_data.copy_type);
+
+		if (kpb->force_copy_type >= COMP_COPY_NORMAL)
```

Comment:
Do we know for a fact that COMP_COPY_NORMAL is the minimum value?

Reply:
@paulstelian97 Yes,

```
enum comp_copy_type {
»       COMP_COPY_NORMAL = 0,»  /**< Normal */                                                                                                                                                             
»       COMP_COPY_BLOCKING,»    /**< Blocking */
»       COMP_COPY_ONE_SHOT,»    /**< One-shot */
};
```

Reply:
I still feel like you should compare with 0 rather than with an enum value. Less than 0 means missing value, 0 or more means we actually have something. (coding style, since of course the generated code is perfectly equivalent).

Reply:
Like this? 
```
 		if (kpb->force_copy_type >= 0)
```

Reply:
Yeah, I'm more on that one. I can accept both but I feel like this one would be better.

Reply:
Ok, fixed.